### PR TITLE
Removing type specific APIs

### PR DESCRIPTION
**Pull request by nucleardreamer**
These type-specific APIs should be left out of the main lib. There should be a better section of examples in the readme or new JS examples to explain and document specific events (`survey-timeout` for example)

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
the corresponding tests need to be removed as well.

> There should be a better section of examples in the readme or new JS examples to explain and document specific events (survey-timeout for example)

Then add them in a PR.

Not sure if we want to keep this method or not. Will mull it over another day.

**Comment by nucleardreamer** [comment]
Ahh, yes indeed I missed that! How about gathering all documentation of type specific APIs into one release? I definitely agree that is a good thing to have them in here, but it should be thorough and documented with some slick examples.

File: lib/index.js [hunk 1]
```diff
@@ -312,26 +312,6 @@ Socket.prototype.setsockopt = function (level, option, value) {
...
-Socket.prototype.survey = function (buf, callback) {
-    if (!this.type === 'surveyor') {
-        throw new Error('Only surveyor sockets can survey.');
-    }
-    var responses = [];
-    function listener (buf) {
-        responses.push(buf);
-    }
-    this.once('survey-timeout', function () {
-        this.removeListener('message', listener);
```
Comment (reqshark) [inline lib/index.js]: where are we setting the `message` event?

**Comment by reqshark** [comment]
+1 :rocket: boom!

type-specific API oversteps into node-nanomsg user concern. 

this stuff was built around strong assumptions of specific user case. I would love to see it gone, as that would be a win for simplicity and predictability

moreover this is undocumented and not implemented correctly. that's very problematic and creates confusion, see #141. 

@nickdesaulniers, the `survey` method deleted here removes an `onmessage` event that was changed to `ondata` when we converted the recv mechanism to streams.

@nucleardreamer, please also remove the `emit` call on line 187 https://github.com/nickdesaulniers/node-nanomsg/blob/master/lib/index.js#L187

guys, i think it would be great to see a small test here based on the outcome of #141 

Thanks @nucleardreamer, this is much needed cleanup!

**Comment by nickdesaulniers** [comment]
if @nucleardreamer removes the extra emit, this patch is good to go. Thanks! :+1: :baseball: :battery:

**Comment by nucleardreamer** [comment]
Nice, Simple things FTW!

**Comment by reqshark** [comment]
thanks! landed in b5533a3b3581f3836c86da664e8d37a5f2ecea00

**Pull request closed without merging**